File: algorithm/38. Count and Say/38.py
class Solution(object):

    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        def getnext(s):
            i=0
            result=""
            s+="k"
            while i<len(s)-1:
                count=1
                if s[i]==s[i+1]:
                    while s[i]==s[i+1]:
                        count+=1
                        i+=1
                    result+=str(count)+s[i]
                    i+=1
                else:
                    result+="1"+s[i]
                    i+=1
            return result
            
        result="1"
        for i in range(1,n):
            result=getnext(result)
        return result
                    

File: algorithm/38. Count and Say/38.cpp
class Solution {
private:
    string getnext(string s){
        string result;  
        int i=0;  
        int count ;  
        while (i < s.size())  
        {  
            count = 1;  
            char temp;  
            if (s[i] == s[i + 1])  
            {  
                while (s[i] == s[i + 1])  
                {  
                    count++;  
                    i++;  
                }  
                temp = count + '0';  
                result = result + temp + s[i];  
                i++;  
            }  
            else  
            {  
                temp = '1';  
                result = result + temp + s[i];  
                i++;  
            }  
        }  
        return result;  
    }
public:
    string countAndSay(int n) {
        string result="1";
        for (int i=1; i<n;i++)
        {
            result=getnext(result);
        }
        return result;
    }
};
